clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/view/ChangeCaptureViewWriter.java

```diff
@@ -190,8 +186,7 @@ synchronized private void initializeVeniceWriter() {
     if (veniceWriter != null) {
       return;
     }
-    veniceWriter =
-        new VeniceWriterFactory(props, pubSubProducerAdapterFactory, null).createVeniceWriter(buildWriterOptions());
+    veniceWriter = new VeniceWriterFactory(props, null).createVeniceWriter(buildWriterOptions());
```

Comment:
same: Let’s avoid creating factories at individual call sites. Instead, we should create the factory
once per server component and reuse it.